mlos_bench/mlos_bench/tests/services/remote/azure/azure_services_test.py

```diff
@@ -6,17 +6,69 @@
 Tests for mlos_bench.services.remote.azure.azure_services
 """
 
+import json
 from copy import deepcopy
+from io import BytesIO
 from unittest.mock import MagicMock, patch
 
 import pytest
+import urllib3
+import requests.exceptions as requests_ex
 
 from mlos_bench.environments.status import Status
 
 from mlos_bench.services.remote.azure.azure_auth import AzureAuthService
 from mlos_bench.services.remote.azure.azure_services import AzureVMService
 
 
+def make_httplib_json_response(status: int, json_data: dict) -> urllib3.HTTPResponse:
+    """
+    Prepare a json response object for use with urllib3
+    """
+    data = json.dumps(json_data).encode("utf-8")
+    response = urllib3.HTTPResponse(
+        status=status,
+        body=BytesIO(data),
+        preload_content=False,
+    )
+    return response
+
+
+@pytest.mark.parametrize(
+    ("total_retries", "operation_status"), [
+        (2, Status.SUCCEEDED),
+        (1, Status.FAILED),
+        (0, Status.FAILED),
+    ])
+@patch("urllib3.connectionpool.HTTPConnectionPool._get_conn")
+def test_wait_host_deployment_retry(mock_getconn: MagicMock,
+                                    total_retries: int,
+                                    operation_status: Status,
+                                    azure_vm_service: AzureVMService) -> None:
+    """
+    Test retries of the host deployment operation.
+    """
+    params = {
+        "pollInterval": 0.1,
+        "requestTotalRetries": total_retries,
+        "deploymentName": "TEST_DEPLOYMENT1",
+        "subscription": "TEST_SUB1",
+        "resourceGroup": "TEST_RG1",
+    }
+
+    # Simulate intermittent connection issues
+    mock_getconn.return_value.getresponse.side_effect = [
+        make_httplib_json_response(200, {"properties": {"provisioningState": "Running"}}),
+        requests_ex.ConnectionError("Connection aborted", OSError(107, "Transport endpoint is not connected")),
+        requests_ex.ConnectionError("Connection aborted", OSError(107, "Transport endpoint is not connected")),
+        make_httplib_json_response(200, {"properties": {"provisioningState": "Running"}}),
+        make_httplib_json_response(200, {"properties": {"provisioningState": "Succeeded"}}),
+    ]
+
+    (status, _) = azure_vm_service.wait_host_deployment(params, is_setup=True)
```

Comment:
I'd move `params` closer to where we use it:
```suggestion
    (status, _) = azure_vm_service.wait_host_deployment(
        params={
            "pollInterval": 0.1,
            "requestTotalRetries": total_retries,
            "deploymentName": "TEST_DEPLOYMENT1",
            "subscription": "TEST_SUB1",
            "resourceGroup": "TEST_RG1",
        },
        is_setup=True
    )
```